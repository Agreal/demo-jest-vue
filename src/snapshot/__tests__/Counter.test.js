import {mount} from '@vue/test-utils'
import Counter from '../Counter.vue'

describe('Counter', () => {

  test('should render correct markup when mount', () => {
    const wrapper = mount(Counter);
    expect(wrapper.vm.count).toBe(0);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('should render correct markup after click button', () => {
    const wrapper = mount(Counter);
    wrapper.find('button').trigger('click');
    expect(wrapper.vm.count).toBe(1);
    expect(wrapper.element).toMatchSnapshot();
  });
});
