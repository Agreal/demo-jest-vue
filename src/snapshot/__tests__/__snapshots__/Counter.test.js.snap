// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correct markup after click button 1`] = `
<div>
  <span
    class="count"
  >
    1
  </span>
   
  <button>
    Increment
  </button>
</div>
`;

exports[`Counter should render correct markup when mount 1`] = `
<div>
  <span
    class="count"
  >
    0
  </span>
   
  <button>
    Increment
  </button>
</div>
`;
